### ISSUE TITLE
[prefer-string-starts-ends-with] Should only complain about .slice() with negative offset

### ISSUE THREAD

**Issue by felixfbecker** [typescript-eslint/typescript-eslint#1690]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-string-starts-ends-with": "error"
  }
}
```

```TS
new URL('https://foo?bar').search.slice(1) === 'bar'
```

**Expected Result**

This is needed to slice off the `?` and not the same as using `.endsWith('bar')`, because `?foobar` would match that too.

**Actual Result**

Error: `Use the 'String#endsWith' method instead.`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `X.Y.Z` |
| `TypeScript`                       | `X.Y.Z` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
Looking at the rule's code, it seems like the intention was entirely that `slice(<non-negative number literal>)` is not intended to be caught by the rule, it's just that it doesn't check if the number literal is a negative one or not.

https://github.com/typescript-eslint/typescript-eslint/blob/e5db36f140b6463965858ad4ed77f71a9a00c5a7/packages/eslint-plugin/src/rules/prefer-string-starts-ends-with.ts#L538-L541
